commit 177ea5b23d2888ac5ca5a589bcd38309163756a2
Author: q66 <q66@chimera-linux.org>
Date:   Sun Feb 15 12:22:38 2026 +0100

    disable fortify for skia
    
    We now encounter the same issue as in webkit with this version,
    so do the same thing as in webkit and disable fortify for now.
    
    This needs to be investigated but probably through webkit first
    as debugging chromium is a pain.

diff --git a/skia/BUILD.gn b/skia/BUILD.gn
index aa009f9..020efa8 100644
--- a/skia/BUILD.gn
+++ b/skia/BUILD.gn
@@ -216,6 +216,18 @@ config("skia_library_config") {
       "SK_DEFAULT_FONT_CACHE_COUNT_LIMIT=256",
     ]
   }
+
+  # Disable fortify because there is a memcpy issue somewhere in initialization,
+  # example backtrace:
+  #
+  # * thread #1, name = 'chromium', stop reason = signal SIGILL: illegal operand
+  #   * frame #0: 0x00005555577615a0 chromium`memcpy at string.h:0 [inlined]
+  #     frame #1: 0x000055555776155d chromium`reset at SkDescriptor.cpp:220:5 [inlined]
+  #     frame #2: 0x0000555557761515 chromium`::SkAutoDescriptor() at SkDescriptor.cpp:156:15
+  #     frame #3: 0x0000555557871af2 chromium`SkStrikeSpec at SkStrikeSpec.cpp:28:15 [inlined]
+  #
+  # We encountered this in webkit before.
+  cflags = [ "-U_FORTIFY_SOURCE" ]
 }
 
 # Split out into separate target so that path_bridge can depend on the public
